rent_golem/backend/rent_golem/clusters/views.py

```diff
@@ -1,9 +1,40 @@
-from rest_framework import viewsets
+from rest_framework import viewsets, status
+from rest_framework.response import Response
 
-from .models import Clusters
-from .serializers import ClustersSerializer
+from rest_framework.permissions import IsAuthenticated
 
+from .models import Cluster
+from .serializers import ClusterSerializer
 
-class ClustersViewSet(viewsets.ModelViewSet):
-    queryset = Clusters.objects.all()
-    serializer_class = ClustersSerializer
+
+class ClusterViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    http_method_names = ['get', 'post', 'head', 'patch', 'delete']
+
+    queryset = Cluster.objects.all()
+    serializer_class = ClusterSerializer
+
+    def destroy(self, request, *args, **kwargs):
+        cluster = Cluster.objects.filter(uuid=kwargs["pk"]).first()
+        if request.user.is_authenticated:
+            if cluster.owner == request.user:
+                cluster.delete()
+                return Response(status=status.HTTP_204_NO_CONTENT)
+            return Response(status=status.HTTP_403_FORBIDDEN)
+        return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+    def update(self, request, *args, **kwargs):
+        cluster = Cluster.objects.filter(uuid=kwargs["pk"]).first()
+
+        if request.user.is_authenticated:
+            if cluster.owner == request.user:
+                if set(request.data.keys()) != {'size'}:
+                    return Response(
+                        status=status.HTTP_403_FORBIDDEN,
+                        data={"message": "You can only change 'size' parameter."}
+                    )
+                cluster.size = request.data['size']
+                cluster.save()
+                return Response(status=status.HTTP_202_ACCEPTED, data=request.data)
+            return Response(status=status.HTTP_403_FORBIDDEN)
```

Comment:
1. It's better to use if + else here for readability.
2. Validation and saving is serializer's job so let's use one.

```
if cluster.owner == request.user:
   serializer = ClusterUpdateSerializer(data=request.data)
   serializer.is_valid(raise_exceptions=True)
   serializer.save()
else:
    return Response(status=status.HTTP_403_FORBIDDEN)
    
class ClusterUpdateSerializer(ModelSerializer):
    class Meta:
        model = Cluster
        fields = ["size"]
```